Objects/moduleobject.c

```diff
@@ -701,6 +701,27 @@ module_getattro(PyModuleObject *m, PyObject *name)
     return NULL;
 }
 
+static PyObject *
+module_str(PyModuleObject *m)
+{
+    PyObject *name, *str;
+
+    name = PyModule_GetNameObject((PyObject *)m);
+    if (name == NULL) {
+        /* XXX PyUnicode_FromStringAndSize is documented as part of
+           the legacy API   */
```

Comment:
Really? I never heart that `PyUnicode_FromStringAndSize` is a legacy API. This is the main way of creating strings in C. Could you point the place in the documentation where it declared legacy?

Reply:
Thanks for the comment.  I don’t have the original Mercurial clone on this laptop, so I can’t look for my original commit.  This code is removed in my latest changes (not pushed yet).